ansible/roles/reboot/tasks/main.yml

```diff
@@ -0,0 +1,27 @@
+---
+# .. vim: foldmarker=[[[,]]]:foldmethod=marker
+
+# Copyright (C) 2022 Julien Lecomte <julien@lecomte.at>
+# Copyright (C) 2022 DebOps <https://debops.org/>
+# .. SPDX-License-Identifier: GPL-3.0-only
+
+- name: Check if reboot is required
+  ansible.builtin.stat:
+    path: '/var/run/reboot-required'
+    get_checksum: no
+  register: reboot_required
+
+- name: Is reboot required?
+  ansible.builtin.debug:
+    var: reboot_required.stat.exists
+
+- name: Reboot DebOps hosts
+  ansible.builtin.reboot:
+    search_paths:
+      - '/lib/molly-guard'
+      - '/sbin'
+      - '/usr/sbin'
+      - '/usr/local/sbin'
```

Comment:
The `search_paths` could be exposed via default variables to allow users to modify them if necessary.